github-workflow-roles/main.tf

```diff
@@ -5,25 +5,44 @@
 locals {
   aws_account_id = var.aws_account_id != "" ? var.aws_account_id : data.aws_caller_identity.current.account_id
 
-  role_name        = coalesce(var.role_name, "github-workflow-role-${var.repository_name}")
-  role_description = "Github Workflow Role for ${var.github_organization_name}/${var.repository_name}"
+  role_name        = var.principal_type == "github" ? coalesce(var.role_name, "github-workflow-role-${var.repository_name}") : var.role_name
+  role_description = var.principal_type == "github" ? "Github Workflow Role for ${var.github_organization_name}/${var.repository_name}" : "IAM role created by Role Vending Machine"
```

Comment:
The description here isn't very descriptive. A pointer back to RVM is good but doesn't give at-a-glance info about the role's purpose.

Reply:
My preference would be to allow the module's caller to specify a role_description variable value and use that -- if not provided, default to what you've got written already.

Reply:
Had to introduce a new variable (role_description) and changed logic to check for that variable.